cmd/curio/guidedsetup/shared.go

```diff
@@ -256,3 +265,166 @@ func ensureEmptyArrays(cfg *config.CurioConfig) {
 		cfg.Apis.ChainApiInfo = []string{}
 	}
 }
+
+func cidPtrToStrptr(c *cid.Cid) *string {
+	if c == nil {
+		return nil
+	}
+	s := c.String()
+	return &s
+}
+
+func coalescePtrs[A any](a, b *A) *A {
+	if a != nil {
+		return a
+	}
+	return b
+}
+
+func MigrateSectors(ctx context.Context, maddr address.Address, mmeta datastore.Batching, db *harmonydb.DB, logMig func(int)) error {
+	mid, err := address.IDFromAddress(maddr)
+	if err != nil {
+		return xerrors.Errorf("getting miner ID: %w", err)
+	}
+
+	sts := statestore.New(namespace.Wrap(mmeta, datastore.NewKey(sealing.SectorStorePrefix)))
+
+	var sectors []sealing.SectorInfo
+	if err := sts.List(&sectors); err != nil {
+		return xerrors.Errorf("getting sector list: %w", err)
+	}
+
+	logMig(len(sectors))
+
+	migratableState := func(state sealing.SectorState) bool {
+		switch state {
+		case sealing.Proving, sealing.Available, sealing.Removed:
+			return true
+		default:
+			return false
+		}
+	}
+
+	unmigratable := map[sealing.SectorState]int{}
+
+	for _, sector := range sectors {
+		if !migratableState(sector.State) {
+			unmigratable[sector.State]++
+			continue
+		}
+	}
+
+	if len(unmigratable) > 0 {
+		fmt.Println("The following sector states are not migratable:")
+		for state, count := range unmigratable {
+			fmt.Printf("  %s: %d\n", state, count)
+		}
+
+		if os.Getenv("CURIO_SKIP_UNMIGRATABLE_SECTORS") != "1" {
```

Comment:
nit: If possible can we not use ENV vars for guided-setup. It should stay as the first step without any pre-requisites if possible.

Reply:
Makes sense, can try to change this over to interactive for guided setup